### PR TITLE
[Enhancement] Remove struct subfield colon separator and change struct subfield equals() behavior

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by Smith-Cruise** [comment]
run starrocks_fe_unittest

**Comment by Smith-Cruise** [comment]
run starrocks_fe_unittest

**Comment by Smith-Cruise** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 1]
```diff
@@ -2050,16 +2050,16 @@ mapType
     : MAP '<' type ',' type '>'
     ;
 
-columnNameColonType
-    : identifier ':' type comment?
+columnNameType
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:2053]: `fieldDesc`

File: fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java [hunk 2]
```diff
@@ -26,10 +26,13 @@
  * comments of struct fields. We set comment to null to avoid compatibility issues.
  */
 public class StructField {
+    // If name is null, that means this StructFiled is unnamed.
     @SerializedName(value = "name")
     private final String name;
     @SerializedName(value = "type")
     private final Type type;
+
+    // comment is not used now, it's always null.
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java:35]: just remove it

File: fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java [hunk 3]
```diff
@@ -122,14 +125,24 @@ public int hashCode() {
...
             return false;
         }
         StructField otherStructField = (StructField) other;
         if (name == null) {
-            return otherStructField.name == null && otherStructField.type.equals(type);
+            // If this() is unnamed struct field, other struct field must also be an unnamed struct field.
+            return otherStructField.name == null && type.equals(otherStructField.type);
+        }
+
+        // Both are named struct field
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java:142]: put it on LOC147

**Review by DorianZheng — COMMENTED** [review]
left some comment, rest LGTM

File: fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java (hunk 2 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java:35]: `comment` field has existed in thrift for a long time, we should keep it.

File: fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java (hunk 3 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/catalog/StructField.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|3|3|100.00%|[]|

**Review by DorianZheng — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructField.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 (hunk 1 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4]: fixed

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
@mergify backport branch-2.5

**Pull request merged**